### DISCUSSION TITLE
More training needed? Less?

### DISCUSSION THREAD

**Discussion by ibrewster** [blakeblackshear/frigate#19228] [Ask A Question] [answered]
I captured the attached image from frigate, in which it identified a single, stationary, car as two cars - and according to the MQTT event that accompanied this image, a moving car (to my understanding that would be the car with area 536355). Interestingly, when I upload this image to frigate+, the suggestions tab shows a single car, properly bounded. 

What I'm wondering is does this mean the base model is doing a better job of identifying the car than my custom trained model? Have I done a bad job of submitting and tagging images, such that I've made things worse? Or is it just random, and perhaps some additional submissions and trainings are the way to go here? Or maybe I'm just overthinking everything? 

![04-12-53_objdetect](https://github.com/user-attachments/assets/85d37058-d8ea-41be-86d2-cd3a00b5cf54)
[1753013573.059158-pscku0.json](https://github.com/user-attachments/files/21336287/1753013573.059158-pscku0.json)

**Comment by inkeep-ai-assistant** [comment]
The scenario you're describing suggests your Frigate+ model may need additional training. When the base Frigate+ model correctly identifies a single car in the suggestions tab but your trained model detects two cars, this indicates the training data may not be optimal for your specific scene[(1)](https://github.com/blakeblackshear/frigate/issues/12540).

For false positives like this, you should submit the image to Frigate+ and label only the single car that's actually present[(1)](https://github.com/blakeblackshear/frigate/issues/12540). The correct approach is to mark the vehicle as "this is a car" and then in Frigate+, draw a bounding box only around the actual car, not the false detection[(1)](https://github.com/blakeblackshear/frigate/issues/12540).

When submitting images for training, you should label the car and draw a box on all parts that are visible within the camera frame[(2)](https://github.com/blakeblackshear/frigate/discussions/9982). If objects extend beyond the frame, only label the visible portions[(2)](https://github.com/blakeblackshear/frigate/discussions/9982).

The fact that the base model performs better in this case suggests your training submissions may have included similar false positives or unclear labeling. Additional training with properly labeled images should improve your model's accuracy for this specific scenario.

📚 **Sources:**
1. [Use Frigate+ to Identify false sublabels like you can with labels](https://github.com/blakeblackshear/frigate/issues/12540)
2. [[Question]: Frigate+ Image submission- labeling an object extending beyond the camera frame](https://github.com/blakeblackshear/frigate/discussions/9982)

**Reply by hawkeye217** [reply, marked as answer]
The AI is mostly correct here - the "base model" is a different model than the "suggestions model", which is a more accurate but much slower model used for generating suggestions on the Frigate+ website. 

When you first set up Frigate+, did you adjust your scores per the docs?

https://docs.frigate.video/plus/first_model#step-4-adjust-your-object-filters-for-higher-scores

**Reply by ibrewster** [reply]
> When you first set up Frigate+, did you adjust your scores per the docs?

I did, and I've adjusted them several times since then. This is just something I've noticed since the latest training on the new model - actually, this is the second training I've done on the new model, the first one was so jumpy on this car that I actually went back to the training I had done on a previous base model (2025.0). Submitting a couple more images and retraining stabilized it significantly, though as you can see it didn't fix it 100%. That said, It does make sense that I might need to tweak the levels a bit with every new training .

It also makes sense that the suggestions model would be better - and totally explains the results as well. Thanks!

**Reply by hawkeye217** [reply]
Happy to help. Let us know if you have any other issues!

**Comment by NickM-27** [comment]
Only you can say whether you've properly annotated. It is possible if there are a number of partial car boxes in your training data it would cause this problem. Otherwise, you may need to do some additional training.

It's important to remember that the suggestions model is not the base model, but a separate model trained specifically for being good at suggestions. It runs at a 640x640 resolution

**Reply by ibrewster** [reply]
> if there are a number of partial car boxes in your training data it would cause this problem

which there are, due to cars being partially out-of-frame, but me annotating them as cars anyway. So perhaps I _haven't_ properly annotated, and need to go back and remove the "partial" images for a better training set? I did the initial annotations as "I know this is a car, so I'll mark it as a car", but perhaps I should have been thinking more along the lines of "This is a good representation of a car in my scene"

**Reply by hawkeye217** [reply]
I've found personally that as long as I've annotated enough cars in my driveway, partial cars elsewhere don't seem to affect it. But this may not be representative of everyone because of different scenes, cameras, etc.

**Reply by NickM-27** [reply]
odds are the cars you are annotating partially are a very different perspective of the car, so there is very little likelihood that those would affect each other
